Sources/SwiftFrameCore/Workers/ImageWriter.swift

```diff
@@ -22,22 +22,24 @@ public final class ImageWriter {
 
         let workGroup = DispatchGroup()
 
-        // Writing images asynchronously gave a big performance boost, what a surprise
-        // Also, since we checked beforehand if the directory is writable, we can safely put of the rendering work to a different queue
-        workGroup.enter()
-        DispatchQueue.global(qos: .userInitiated).ky_asyncOrExit {
-            try ImageWriter.write(images: slices, to: outputPaths, locale: locale, suffix: suffix, format: format)
-            workGroup.leave()
-        }
-
-        if outputWholeImage {
+        suffixes.forEach { suffix in
```

Comment:
One thought: If this is just running the same code again with a different file suffix, could we only generate the image once and then just copy the file to the other suffixes? That might be a lot faster.

Reply:
That's an excellent idea!

Reply:
Actually, I guess that's what we're already doing. We render the big image in the first step of the method and then slice it up before entering any work group/async context
```swift
guard let image = context.cg.makeImage() else {
    throw NSError(description: "Could not render output image")
}
let slices = try sliceImage(image, with: sliceSize, gapWidth: gapWidth)
```